src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,6 +235,40 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (Options.UsePostAuthorizationRequests)
+            {
+                using (var buffer = new MemoryStream())
+                using (var writer = new StreamWriter(buffer))
+                {
+                    var inputs = new StringBuilder();
+                    foreach (var parameter in message.Parameters)
+                    {
+                        var name = Options.HtmlEncoder.HtmlEncode(parameter.Key);
+                        var value = Options.HtmlEncoder.HtmlEncode(parameter.Value);
+
+                        inputs.AppendFormat(CultureInfo.InvariantCulture, InputTagFormat, name, value);
+                    }
+
+                    var issuer = Options.HtmlEncoder.HtmlEncode(message.IssuerAddress);
+
+                    writer.Write(string.Format(CultureInfo.InvariantCulture, AuthorizationFormFormat, issuer, inputs));
```

Comment:
writer.Write is only called once. Why not just call Encoding.UTF8.GetBytes directly? Then you don't need the StreamWriter, the MemoryStream, etc..